frame/scheduler/src/lib.rs

```diff
@@ -15,38 +15,62 @@
 //!
 //! ## Overview
 //!
-//! This Pallet exposes capabilities for scheduling dispatches to occur at a
-//! specified block number or at a specified period. These scheduled dispatches
+//! This Pallet exposes capabilities for scheduling runtime calls to occur at a
+//! specified block number or at a specified period. These scheduled runtime calls
 //! may be named or anonymous and may be canceled.
 //!
-//! **NOTE:** The scheduled calls will be dispatched with the default filter
-//! for the origin: namely `frame_system::Config::BaseCallFilter` for all origin
-//! except root which will get no filter. And not the filter contained in origin
-//! use to call `fn schedule`.
+//! **NOTE:** Instead of the filter contained in the origin used to call `fn schedule`, scheduled runtime calls will be 
+//! dispatched with the default filter for the origin: namely `frame_system::Config::BaseCallFilter` for all origin
+//! types (except root which will get no filter).
 //!
 //! If a call is scheduled using proxy or whatever mecanism which adds filter,
-//! then those filter will not be used when dispatching the schedule call.
+//! then those filter will not be used when dispatching the schedule runtime call.
 //!
-//! ## Interface
+//! ## Pallet API
 //!
-//! ### Dispatchable Functions
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events and errors.
 //!
-//! * `schedule` - schedule a dispatch, which may be periodic, to occur at a specified block and
-//!   with a specified priority.
-//! * `cancel` - cancel a scheduled dispatch, specified by block number and index.
-//! * `schedule_named` - augments the `schedule` interface with an additional `Vec<u8>` parameter
-//!   that can be used for identification.
-//! * `cancel_named` - the named complement to the cancel function.
-
+//! ### Warning
+//!
+//! This Pallet executes all scheduled runtime calls in the
+//! [`frame_support::traits::Hooks::on_initialize`] hook.
+//!
+//! **Do NOT schedule runtime calls if they satisfy any of the following points:**
+//!
+//! * [ ] Runtime call does not need to be executed at a specific block (not a hard requirement).
+//! 	Use [`frame_support::traits::Hooks::on_idle`] hook for runtime calls that can be executed
+//!   after some block but do not need to be executed at a specific block.
+//! * [ ] No unbounded iterations.
+//! * [ ] Runtime call weight exceeds configured [`Config::MaximumWeight`].
+//! 	If you are scheduling a heavy weight runtime call, make sure to set the `MaximumWeight` type
+//! according to your chain's needs. You can find the weight conversions to real time from these
+//! [`sp_weights::constants`].
+//!
+//! Please be aware that any scheduled runtime calls executed in a future block may __fail__ or may
+//! result in __unintended behavior__ since the runtime could have upgraded between the time of
```

Comment:
```suggestion
//! result in __undefined behavior__ since the runtime could have upgraded between the time of
```

Sounds like this would really nail the point home.

Reply:
Agreed.